staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -192,8 +200,10 @@ type CustomResourceDefinition struct {
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
 	// Spec describes how the user wants the resources to appear
+	// +optional
```

Comment:
I'm confused why spec is omitempty...

Reply:
I don't think this is right.

Reply:
> I don't think this is right.

ok to remove the omitempty? I hardly think there are CRDs out there without `.spec` but technically this would be a breaking change.

/cc @liggitt

Reply:
if no CRD submitted to the server without spec present would have ever succeeded (and it wouldn't, our validation requires fields under spec), then it's not a breaking change.

Reply:
Fixed.